src/metabase/models/recent_views.clj

```diff
@@ -341,22 +340,10 @@
                                            :archived :location :trashed_from_location]
                                   :where [:and
                                           [:in :id collection-ids]
-                                          [:= :archived false]]})
-          coll->parent-id (fn [c]
-                            (some-> c collection/effective-location-path collection/location-path->ids last))
-          parent-ids (into #{} (keep coll->parent-id) collections)
-          id->parent-coll (merge {nil (root-coll)}
-                                 (when (seq parent-ids)
-                                   (t2/select-pk->fn identity :model/Collection
-                                                     {:select [:id :name :authority_level]
-                                                      :where [:in :id parent-ids]})))]
-      ;; replace the collection ids with their collection data:
-      (map
-       (fn effective-collection-assocer [c]
-         (assoc c
-                :effective_parent (->> (coll->parent-id c) id->parent-coll)
-                :effective_location (collection/effective-location-path c)))
-       collections))))
+                                          [:= :archived false]]})]
+      (->> (t2/hydrate collections :effective_parent)
+
+           (map #(m/dissoc-in % [:effective_parent :type]))))))
```

Comment:
`:type` is expected in the `:collection` field of search results but not in recent views, so I'm just removing it here. It probably wouldn't hurt to include it here too but it breaks a lot of tests.